Dockerfile

```diff
@@ -0,0 +1,41 @@
+# Build stage
+FROM mcr.microsoft.com/dotnet/sdk:7.0.400-bookworm-slim AS build
+WORKDIR /build
+
+COPY *.sln ./
+COPY **/*.csproj ./
+
+RUN dotnet sln list | grep ".csproj" \
+    | while read -r line; do \
+    mkdir -p $(dirname $line); \
+    mv $(basename $line) $(dirname $line); \
+    done;
```

Comment:
What does this do?

Reply:
The wildcard copy command before this doesn’t keep the original directory structure of the .csproj files. This just restores the original directory structure. Refresh.GameServer.csproj will get moved to Refresh.GameServer/Refresh.GameServer.csproj. This step happens before copying any actual code because in the docker build cache this step doesn’t have to be rerun unless any of the .csproj files change. If you copied the entire repo first then any change would trigger a cache invalidation. You could alternatively copy the project files manually and specify the folder they should be in.